### PR TITLE
AO3-6006 Show "pseud (username)" in dashboard heading

### PR DESCRIPTION
**Pull request by tlee911**
# Pull Request Checklist

* [x] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [x] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [x] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [x] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [x] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6006

## Purpose

Make it clear on user dashboard that a pseud is being displayed by showing the username as well if a non-default pseud is being displayed.  Note that I have removed the links from the heading here.  I felt they weren't useful since you were already viewing this particular user's page.

## Testing Instructions

1. Go to user page of user with extra pseuds
2. Verify that if default pseud is selected only username is displayed in the heading
3. Verify that if a different pseud is selected, the byline format "pseud (username)" is displayed in the heading

New test added in `user_dashboard.feature` to account for this, note that I added examples to also check works and bookmarks pages for the same thing (didn't see tests for this elsewhere).  Also, this can easily be expanded with one more example for the series page (AO3-6007).

## References

See also AO3-6007 and #3887.  Same scenario outline is used for testing, since the expected outcome is similar.  There may be conflicts with that one depending on the merge order.

## Credit

tlee911 (he/him)

### CODE REVIEW

File: app/views/users/_header.html.erb [hunk 1]
```diff
@@ -1,10 +1,9 @@
 <!--Descriptive page name, messages and instructions-->
 <div class="primary header module">
-	<% 	display_name = @pseud ? @pseud.name : @user.login	%>
-	<h2 class="heading">
-	  <%= @pseud ? link_to_unless_current(display_name, user_pseud_path(@user, @pseud)) : link_to_unless_current(display_name, @user) %>
```
Comment (sarken) [inline app/views/users/_header.html.erb:5]: Just noting that it looks like this was only ever _actually_ displayed as a link on a user's profile page, so I don't think `link_to_unless_current` offered much value, considering the "Dashboard" link it right there.

We could go back to using a link if other people think it's helpful, though.

**Comment by sarken** [comment]
Could you split this into separate pull requests, please? For ease of testing and review, we strongly prefer one pull request per issue unless the code for the two issues is difficult to untangle.

**Comment by tlee911** [comment]
> Could you split this into separate pull requests, please? For ease of testing and review, we strongly prefer one pull request per issue unless the code for the two issues is difficult to untangle.

Yeah that's fair.  I started AO3-6007 thinking it was going to be a very similar one-liner, but when all is said and done with the counts and everything, better to keep them separate.

**Comment by sarken** [comment]
Thanks for understanding! It looks good to me, so I'll mark this as approved once Travis finishes running and shows the nice green check.

**Pull request merged**